### PR TITLE
Remove compat bellow 1.0

### PR DESCRIPTION
**Pull request by StorytellerCZ**
Re-creating #11226 with cherry picked commits.

### CODE REVIEW

**Comment by filipenevola** [comment]
@StorytellerCZ could you add to the history all the removals?

**Comment by StorytellerCZ** [comment]
@filipenevola Sure thing!

**Comment by StorytellerCZ** [comment]
The tests are now failing due to package versions being set for 2.3-beta.1.

File: History.md [hunk 1]
```diff
@@ -1,5 +1,36 @@
 ## v2.3.0, UNRELEASED
 
+### Breaking changes
+* Removed deprecated `mobile-port` flag
```
Comment (filipenevola) [inline History.md:4]: these 3 should be under `meteor-tool@2.3`

**Review by filipenevola — CHANGES_REQUESTED** [review]

File: History.md (hunk 1 above)
Comment (StorytellerCZ) [reply History.md:49]: Done

**Pull request merged**